### PR TITLE
Implement index:stats

### LINKED ISSUE #608: Implement API action index:stats
Implement API action index:stats

### PR DESCRIPTION
**Pull request by Leodau**
## What does this PR do?
This **PR** implements the recently added [index:stats](https://docs.kuzzle.io/core/2/api/controllers/index/stats/) _(@ Kuzzle Backend v2.10.0)_
So it can be used directly, instead of having to use **query** to access it.

#### Todo:

- [x] Implementation
- [x] Unit Testing
- [x] Doc + Code Snippet

### How should this be manually tested?

1. Launch a Kuzzle stack.
2. Directly use `sdk.index.stats()`
3. Verify that you get stats that make sense.

Closes #608 

> :information_source: It seems that **Codecov** is not taking the latest commit? :facepalm:

### CODE REVIEW

**Review by scottinet — APPROVED** [review]

File: doc/7/controllers/index/stats/snippets/stats.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const stats = await kuzzle.index.stats();
+
+console.log(JSON.stringify(stats));
```
Comment (Aschen) [inline doc/7/controllers/index/stats/snippets/stats.js:3]: Can you add the JSON structure in comment please?

So user know how to manipulate the result then

**Review by Aschen — CHANGES_REQUESTED** [review]

File: src/controllers/Index.ts [hunk 2]
```diff
@@ -99,4 +99,19 @@ export class IndexController extends BaseController {
...
+
+  /**
+   * Returns detailed storage usage statistics.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/index/stats/
+   *
+   * @param options Additional options
+   *    - `queuable` If true, queues the request during downtime, until connected to Kuzzle again
+   */
+  stats (options: { queuable?: boolean } = {}): Promise<any> {
```
Comment (Aschen) [inline src/controllers/Index.ts:111]:
```suggestion
  stats (options: { queuable?: boolean } = {}): Promise<JSONObject> {
```

**Review by Aschen — APPROVED** [review]

File: src/controllers/Index.ts (hunk 2 above)
Comment (Leodau) [reply src/controllers/Index.ts:111]: Oh good catch

**Review by alexandrebouthinon — COMMENTED** [review]
To fix unit tests, you need to add this line at the top of `src/controllers/Index.ts`:
```js
import { JSONObject } from '../types';
```

**Review by alexandrebouthinon — APPROVED** [review]

**Pull request merged**